File: leetcode_169/solution.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count, major = 1, nums[0]
        for i in range(1, len(nums)):
            if count == 0:
                count += 1
                major = nums[i]
            else:
                if nums[i] != major:
                    count -= 1
                else:
                    count += 1
        return major

File: leetcode_169/solution.go
package main

//func majorityElement(nums []int) int {
//	count, major := 1, nums[0]
//	for i := 1; i < len(nums); i++ {
//		if count == 0 {
//			major = nums[i]
//			count++
//		} else {
//			if nums[i] == major {
//				count++
//			} else {
//				count--
//			}
//		}
//	}
//	return major
//}

func majorityElement(nums []int) int {
	count, major := 1, nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] == major {
			count++
		} else {
			if count != 0 {
				count--
			} else {
				count, major = 1, nums[i]
			}
		}
	}
	return major
}